packages/airnode-admin/src/implementation.ts

```diff
@@ -83,7 +105,11 @@ export async function deriveSponsorWalletAddress(airnodeXpub: string, airnodeAdd
 }
 
 export async function sponsorRequester(airnodeRrp: AirnodeRrp, requesterAddress: string, overrides?: ethers.Overrides) {
-  await airnodeRrp.setSponsorshipStatus(requesterAddress, true, overrides ?? {});
+  await airnodeRrp.setSponsorshipStatus(
+    requesterAddress,
+    true,
+    parseOverrides(await airnodeRrp.provider.getFeeData(), overrides)
```

Comment:
I feel like you could reduce a lot of duplication by supplying the overrides up-front to these functions?
So wherever these functions are called, the caller should be responsible for calling `parseOverrides`?
For example [`const overrides = admin.parseCliOverrides(args);`](https://github.com/api3dao/airnode/blob/230f9112bb8476df0a1d8ab6a2f05421320fc9f6/packages/airnode-admin/src/cli.ts#L207) could be a good place to do this and then it's a small diff.

Reply:
Thanks Aaron, great points to discuss. I probably should have added some context initially to how I wrote it :sweat_smile: : 

> The user shouldn't be specifying both eip1559 and legacy pricing override options, so I think we should error if both are specified? This is how ethers does it internally (see below).

I intentionally allowed the mixed overrides to pass through `parseOverrides` because this approach was also used in [834](https://github.com/api3dao/airnode/pull/834#discussion_r809034853). (Of course, since the `parsedCliOverrides` will now also pass through `parseOverrides` here so I'm happy to add our own checks here.)

> ethers defaults to a fee of maxPriorityFeePerGas = BigNumber.from("2500000000");

Sorry, my bad I should have linked the task [AN-454](https://api3dao.atlassian.net/jira/software/projects/AN/boards/18?selectedIssue=AN-454). 3.1 was specified in the description so I think the same idea for the higher priority fee was intended here.

> I feel like you could reduce a lot of duplication by supplying the overrides up-front to these functions?
So wherever these functions are called, the caller should be responsible for calling parseOverrides?
For example [const overrides = admin.parseCliOverrides(args);](https://github.com/api3dao/airnode/blob/230f9112bb8476df0a1d8ab6a2f05421320fc9f6/packages/airnode-admin/src/cli.ts#L207) could be a good place to do this and then it's a small diff.

Hmm the reason I put it inside the `implementation.ts` methods is so that it works the same way for the CLI and SDK. Otherwise we'd need to expose parseOverrides as an SDK function to the user? 

Happy to make the changes so please let me know what you think.

Reply:
Hey

Thank you for the context and detailed responses.

I see @ "3.1". I don't think we're tied to that value.

My main concern is the if statement, which is complex and difficult to read. I'm mostly concerned with how we can improve the readability of that statement :relaxed:  

That makes sense @ `parseOverrides`. No problem.